packages/sdk/src/types/runtime/assets/assetManager.ts

```diff
@@ -94,6 +97,71 @@ export class AssetManager {
         return createForwardPromise(asset, promise);
     }
 
+    public loadSoundAsset(groupName: string, uri: string, options: LoadSoundOptions): ForwardPromise<string> {
+
+        log.error('app', "STARTING LOADSOUNDASSET");
+        const soundAssetId = UUID();
+        this.enqueueLoadSoundPromise(
+            soundAssetId, {
+                resolve: () => { /* empty */ },
+                reject: () => { /* empty */ },
+            });
+        const promise = this.sendLoadAssetsPayload({
+            type: 'load-sound',
+            id: soundAssetId,
+            uri
+        } as LoadSound)
+            .then<string>(payload => {
+                if (payload.failureMessage || payload.assets.length !== 1) {
+                    this.notifySoundLoaded(soundAssetId, false, payload.failureMessage);
+                    return Promise.reject(`Loading sound asset ${uri} failed: ${payload.failureMessage}`);
+                }
+                // soundAssetId = payload.assets[0].id;
+                this.notifySoundLoaded(soundAssetId, true);
+            });
+        this.registerLoadPromise(promise);
+
+        return createForwardPromise(soundAssetId, promise);
+    }
+
+    public unloadSoundAsset(handle: string) {
+    }
+
+    public loadSoundPromises: { [name: string]: ExportedPromise[] };
+
+    public notifySoundLoaded(soundAssetId: string, success: boolean, reason?: any): void {
+        if (!!this.loadSoundPromises && !!this.loadSoundPromises[soundAssetId]) {
+            const loadSoundPromises = this.loadSoundPromises[soundAssetId].splice(0);
+            delete this.loadSoundPromises[soundAssetId];
+            for (const promise of loadSoundPromises) {
+                if (success) {
+                    promise.resolve();
+                } else {
+                    promise.reject(reason);
+                }
+            }
+        }
+    }
+
+    public enqueueLoadSoundPromise(soundAssetId: string, promise: ExportedPromise): void {
+        if (!this.loadSoundPromises) {
+            this.loadSoundPromises = {};
+        }
+        if (!this.loadSoundPromises[soundAssetId]) {
+            this.loadSoundPromises[soundAssetId] = [];
+        }
+        this.loadSoundPromises[soundAssetId].push(promise);
+    }

```

Comment:
What is the purpose of this API? The load call already returns a promise, do we need alternative mechanisms for waiting for the load? At a minimum, I think these should be private, as ExportedPromise is ironically not a package-exported type.

Reply:
Yes, I believe this is needed. I can make it private, though - good call. The purpose is that while we return a promise that the user can wait on, the user is never required to actually await. If they don't await, the node side will internally wait for the sound to be created before it starts to play. This is similar to the animation system, which internally has to wait on animations to be created before they start animating.

It is different from the material/texture code you wrote, which allows for the material id to be created and assigned right away.